README.MD

```diff
@@ -0,0 +1,114 @@
+```
+
+Each validator receives a `passwordRule` _(number)_ argument in its constructor which defines the number of expected characters associated with the validator. For example, the `const minLength = new MinLengthValidator(10)` defined above means that the passowrd must be atleast __10 characters__ long to be considered valid.
+
+NB: The only exception to this is the `SpaceCharacterValidator` which requires no arguments and simply ensures the password contains no spaces.
+
+### Conflicting Validators
+
+In a case of conflict validations, for example, a password must be __12 characters long minimum__, and also be __8 characters maximum__, we throw a `PasswordValidatorConflictException` error. This is because minimum length cannot be greater than maximum length.
+
+```ts
+import { MinLengthValidator, MaxLengthValidator } from '@password-validator/core';
+
+const pm = new PasswordValidatorManager(); // Create a password validator manager
+
+const minLength = new MinLengthValidator(12);
+const maxLength = new MaxLengthValidator(8);
+
+pm.register(minLength, maxLength);
+
+const result = pm.validate('MyPassword123!*') // Throws PasswordValidatorConflictException --> `minLength cannot be greater than maxLength`
+```
+
+### Validators & Managers Tables
+
+Supported Validators and Managers as of now are:
+
+|     Rules            |               Descriptions                                                                                                       |
+|:---------------------|:---------------------------------------------------------------------------------------------------------------------------------|
+| **Validations**                       |
+|**DigitValidator(passwordRule: number)**   | specifies password must include _passwordRule_ number of digits                         |
+|**LowerCaseValidator(passwordRule: number)**  | specifies password must include at least _passwordRule_ number of lowercase character(s)                     |
+|**UpperCaseValidator(passwordRule: number)**| specifies password must include at least _passwordRule_ number of uppercase character(s)  |
+|**MaxLengthValidator(passwordRule: number)**| specifies password must not exceed _passwordRule_ number of character(s)  |
+|**MinLengthValidator(passwordRule: number)**  | specifies password must not be less than _passwordRule_ number of character(s)                      |
+|**SpecialCharacterValidator(passwordRule: number)**          | specifies password must include at least _passwordRule_ number of special character(s)                                                                |
+|**SpaceCharacterValidator()**   | specifies password must not include spaces                       |
+| **Managers**                       |
+|**PasswordValidatorManager(...validators: PasswordValidator[])**          | Allows registering multiple validators at once to build a more complex validation suite.                                                             |
```

Comment:
Just let them know that validators must be registered before being used for validation. And this is done via the register method of the password validator manager.

The validate method of the validation manager is the interface for validating passwords based on the registered validators.